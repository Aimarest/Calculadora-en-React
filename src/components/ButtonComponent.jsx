

export const ButtonComponent = ({value, action, isDisable}) => {

  const handleClick = () =>{
    action(value);
  }

  return (
    <button onClick={handleClick}  disabled={isDisable} name={value} value={value} className="btn">{value}</button>
  )
}
