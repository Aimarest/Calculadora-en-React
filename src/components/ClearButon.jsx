

export const ClearButon = ({ value, resetResult }) => {

  const handleClick = () => {
    resetResult();
  }
  return (
    <button onClick={handleClick} value={value} className="btnClear btn">{value}</button>
  )
}
