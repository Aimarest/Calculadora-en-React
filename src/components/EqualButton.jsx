

export const EqualButton = ({value, action, isDisable,setIsDisable}) => {


    const handleClick = () =>{
        action(value);
        setIsDisable(true);
        
      }
    
      return (
        <button onClick={handleClick} value={value} disabled={isDisable} className="btn">{value}</button>
      )
}
