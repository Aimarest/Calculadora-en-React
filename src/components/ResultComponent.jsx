

export const ResultComponent = () => {
  return (
    <>
        <h2 className="result">Result</h2>
    </>
  )
}
