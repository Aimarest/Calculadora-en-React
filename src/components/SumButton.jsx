

export const SumButton = ({value,  sum }) => {

    const handleSum = () =>{
        sum(value)
    }
  return (
   <button onClick={handleSum} value={value} className="sumBtn">{value}

   </button>
  )
}
