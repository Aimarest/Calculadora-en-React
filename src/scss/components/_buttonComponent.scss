.buttons {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr 1fr;
    gap: 3px 3px;

    @media screen and (min-width:400px) {

        gap: 10px 15px;

    }

    @media screen and (min-width:764px) {

        gap: 15px 20px;

    }

}

.btn:active {
    background-color: #00916e;
}

.btn {
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 32px;
    padding: 0.6em;
    border: none;
    cursor: pointer;
    min-width: 2.2em;
    font-size: 1em;
    color: white;
    background-color: #00916e;
    box-shadow: 5px 5px 10px -3px #00000040, -5px -5px 15px 3px #00ffc11f;

    @media screen and (min-width:400px) {

        padding: 1.3em
    }

    @media screen and (min-width:764px) {

        padding: 1.8em
    }

    @media screen and (min-width:1024px) {

        padding: 2em
    }

}

.btnClear {
    margin-top: 1em;
    background-color: #224870;
    width: 8em;

    @media screen and (min-width:764px) {

        margin-top: 3em;

    }

    @media screen and (min-width:1024px) {
        width: 20em;
        margin-top: 5em;

    }

}