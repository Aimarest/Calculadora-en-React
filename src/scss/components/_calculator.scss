.calculator {
    margin: 2em;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 40%;
    max-height: 30%;
    max-width: 500px;
    background-color: $calculator-color;
    border-radius: 12px;
    padding: 1.5em 2.5em;
    border: 4px solid $background-color;
    box-shadow: 15px 10px 0 5px $shadow;

    @media screen and (min-width:400px) {
        padding: 4em 3.8em;
    }

    @media screen and (min-width:764px) {
        padding: 6em 2.5em;
    }
}