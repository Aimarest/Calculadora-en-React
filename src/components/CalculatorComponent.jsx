import { ButtonComponent } from './ButtonComponent';
import { ResultComponent } from './ResultComponent';
import {useState} from 'react';

export const CalculatorComponent = () => {
const [text, setText] = useState('');

  const addValueToResult = (value) =>{
    setText((text) => [...text, value + ''])
  }
  return (
    <>
      <ResultComponent text={text} />
      <section className='buttons'>
        <ButtonComponent value={7} addValueToResult={addValueToResult} />
        <ButtonComponent value={8} addValueToResult={addValueToResult}/>
        <ButtonComponent value={9} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'%'} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'Clear'} addValueToResult={addValueToResult}/>
        <ButtonComponent value={4}addValueToResult={addValueToResult} />
        <ButtonComponent value={5} addValueToResult={addValueToResult}/>
        <ButtonComponent value={6} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'X'}addValueToResult={addValueToResult} />
        <ButtonComponent value={'/'} addValueToResult={addValueToResult}/>
        <ButtonComponent value={1} addValueToResult={addValueToResult}/>
        <ButtonComponent value={2} addValueToResult={addValueToResult}/>
        <ButtonComponent value={3} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'+'} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'='} addValueToResult={addValueToResult}/>
        <ButtonComponent value={0} addValueToResult={addValueToResult}/>
      </section>

    </>
  )
}
