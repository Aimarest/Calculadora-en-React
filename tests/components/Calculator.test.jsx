import { CalculatorComponent } from '../../src/components/CalculatorComponent';
import { render, screen, fireEvent } from '@testing-library/react';
describe('Test on CalculatorComponent', () => {
    test('It should display the initial component', () => {
        const { container } = render(
            <CalculatorComponent />
        );
        expect(container).toMatchSnapshot();
    })
    test('if the sum of two and six is made the text of the result must be 8', () => {
        render(<CalculatorComponent />);
        const btnValueTwo = screen.getByRole('button', { name: 2 });
        const btnValueSix = screen.getByRole('button', { name: 6 });
        const btnAddOperator = screen.getByRole('button', { name: '+' });
        const btnEqual = screen.getByRole('button', { name: '=' });
        const textResult = screen.getByText('Result:');
        fireEvent.click(btnValueTwo);
        fireEvent.click(btnAddOperator);
        fireEvent.click(btnValueSix);
        fireEvent.click(btnEqual);
        expect(textResult.innerHTML).toContain('8');
    })
    test('If I do the division of 10 by 5 the result should be 2', () => {
        render(<CalculatorComponent />);
        const btnValueOne = screen.getByRole('button', { name: 1 });
        const btnValueZero = screen.getByRole('button', { name: 0 });
        const btnValueFive = screen.getByRole('button', { name: 5 });
        const btnDivOperator = screen.getByRole('button', { name: '/' });
        const btnEqual = screen.getByRole('button', { name: '=' });
        const textResult = screen.getByText('Result:');
        fireEvent.click(btnValueOne);
        fireEvent.click(btnValueZero);
        fireEvent.click(btnDivOperator);
        fireEvent.click(btnValueFive);
        fireEvent.click(btnEqual);
        expect(textResult.innerHTML).toContain('2');
    })
    test('If I do an operation and hit the clear button, the result should be empty', () => {
        render(<CalculatorComponent />);
        const btnValueOne = screen.getByRole('button', { name: 1 });
        const btnValueZero = screen.getByRole('button', { name: 0 });
        const btnValueFive = screen.getByRole('button', { name: 5 });
        const btnSubOperator = screen.getByRole('button', { name: '-' });
        const btnEqual = screen.getByRole('button', { name: '=' });
        const textResult = screen.getByText('Result:');
        const btnClear = screen.getByText('Clear');
        fireEvent.click(btnValueOne);
        fireEvent.click(btnValueZero);
        fireEvent.click(btnSubOperator);
        fireEvent.click(btnValueFive);
        fireEvent.click(btnEqual);
        fireEvent.click(btnClear);
        expect(textResult.innerHTML).toBe("Result: ");
    })
    test('When the percentage of 8 is done, the result is 0.8', () => {
        render(<CalculatorComponent />);
        const btnValueEight = screen.getByRole('button', { name: 8 });
        const btnPercentOperator = screen.getByRole('button', { name: '%' });
        const btnEqual = screen.getByRole('button', { name: '=' });
        const textResult = screen.getByText('Result:');
        fireEvent.click(btnValueEight);
        fireEvent.click(btnPercentOperator);
        fireEvent.click(btnEqual);
        expect(textResult.innerHTML).toBe("Result: 0.8");
    })
    test('When trying to make a percentage with two numbers, I should get an error in the input', () => {
        render(<CalculatorComponent />);
        const btnValueEight = screen.getByRole('button', { name: 8 });
        const btnPercentOperator = screen.getByRole('button', { name: '%' });
        const btnEqual = screen.getByRole('button', { name: '=' });
        const textError = screen.getByLabelText('errorMessage')
        fireEvent.click(btnValueEight);
        fireEvent.click(btnPercentOperator);
        fireEvent.click(btnValueEight);
        fireEvent.click(btnEqual);
        expect(textError.innerHTML).toBe("Operation not possible, click clear button to continue");
    })
    test('When the button has been pressed without an operation performed an error should be displayed on the input', () => {
        render(<CalculatorComponent />)
        const btnEqual = screen.getByRole('button', { name: '=' });
        const textError = screen.getByLabelText('errorMessage');
        fireEvent.click(btnEqual);
        expect(textError.innerHTML).toBe("You have to enter a valid operation. Please click clear button to continue")
    })
});
