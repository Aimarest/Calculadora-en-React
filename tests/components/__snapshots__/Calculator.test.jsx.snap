// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on CalculatorComponent It should display the initial component 1`] = `
<div>
  <section
    class="calculator"
  >
    <div
      class="input"
    >
      <p
        class="text"
      />
      <p
        aria-label="errorMessage"
        class="errorMessage text"
      />
      <p
        class="text"
      >
        Result: 
         
      </p>
    </div>
    <section
      class="buttons"
    >
      <button
        class="btn"
        name="7"
        value="7"
      >
        7
      </button>
      <button
        class="btn"
        name="8"
        value="8"
      >
        8
      </button>
      <button
        class="btn"
        name="9"
        value="9"
      >
        9
      </button>
      <button
        class="btn"
        name="%"
        value="%"
      >
        %
      </button>
      <button
        class="btn"
        name="4"
        value="4"
      >
        4
      </button>
      <button
        class="btn"
        name="5"
        value="5"
      >
        5
      </button>
      <button
        class="btn"
        name="6"
        value="6"
      >
        6
      </button>
      <button
        class="btn"
        name="X"
        value="X"
      >
        X
      </button>
      <button
        class="btn"
        name="1"
        value="1"
      >
        1
      </button>
      <button
        class="btn"
        name="2"
        value="2"
      >
        2
      </button>
      <button
        class="btn"
        name="3"
        value="3"
      >
        3
      </button>
      <button
        class="btn"
        name="+"
        value="+"
      >
        +
      </button>
      <button
        class="btn"
        name="-"
        value="-"
      >
        -
      </button>
      <button
        class="btn"
        name="/"
        value="/"
      >
        /
      </button>
      <button
        class="btn"
        name="="
        value="="
      >
        =
      </button>
      <button
        class="btn"
        name="0"
        value="0"
      >
        0
      </button>
    </section>
    <button
      class="btnClear btn"
      value="Clear"
    >
      Clear
    </button>
  </section>
</div>
`;
